### ISSUE TITLE
Adding macro and define to write assembly code for both 32/64 bits easier

### ISSUE THREAD

**Issue by zl-wang** [eclipse/omr#483]
We need common assembly names for 32bit and 64bit, in order to write a single copy of code without #if all over the places.
We also need mtvsrwz, since upper 32bit in GPR can be garbage when C++ compiler/JIT compiler optimizes it without keeping loading from memory.   This one is tricky because you even cannot see it in debugger(by definition, only lower 32bit value is shown in 32bit mode).   As a matter of fact, when you debug/step it, it can run successfully.

**Comment by mstoodle** [comment]
@zl-wang Before I close this issue (since the code has been merged), could you please update the title to something like "Add assembler macros to simplify writing code for both 32- and 64-bits". In general, please try to make your titles more specific: "Add macro and define" could be just about anything and affect just about anything, whereas the actual change is much more specific than that, and the title and messages should reflect that.

Thanks, Julian!

**Comment by fjeremic** [comment]
@mstoodle Not quite. See https://help.github.com/articles/closing-issues-via-commit-messages/.

> To close an issue in the same repository, use one of the keywords in the list below followed by a reference to the issue number in the commit message. For example, a commit message with Fixes #483 will close issue 483 in that repository once the commit is merged into the default branch.

We may want to update our guidelines. Perhaps we want to annotate issues like this:

```
Super duper commit subject

Super duper commit body explaining what the change was. It has lots 
of useful information for future readers, etc, etc. It is properly 
formatted and within the column limits.

Fixes #483
Signed-off-by: John Smith john@smith.com
```

Thoughts?

**Comment by mstoodle** [comment]
Thanks for renaming, @zl-wang . Can this issue be closed now, or are there more macros / defines to be added via other commits?

**Comment by zl-wang** [comment]
Sure, close it.  Of course, addition of new requirements can arise in the future.
